hr_contract_cae/models/hr_contract.py

```diff
@@ -1,11 +1,82 @@
 from dateutil.relativedelta import relativedelta
 
-from odoo import api, fields, models
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class ContractType(models.Model):
+
+    _inherit = "hr.contract.type"
+
+    echelon = fields.Selection(
+        [("main", "Main"), ("amendment", "Amendment")],
+        string="Payment Mode",
+        default="main",
+    )
+    max_usage = fields.Integer(
+        string="Maximum Usage", default=1, required=True
+    )
 
 
 class Contract(models.Model):
     _inherit = "hr.contract"
 
+    name = fields.Char(compute="_compute_name")
+    employee_id = fields.Many2one(required=True)
+    state = fields.Selection(copy=False)
+    state_admin = fields.Text(string="Administrative state", copy=False)
+    type_id = fields.Many2one(
+        string="Contract Type",
+        domain="[('echelon', '=', type_echelon)]",
+        default="",
+        copy=False,
+    )  # Todo: rename translation from "Catégorie de l'employé" to "Type de Contrat"
+    type_echelon = fields.Selection(
+        [("main", "Main"), ("amendment", "Amendment")],
+        string="Contract Type Echelon",
+        copy=False,
+        compute="_compute_type_echelon",
+    )  # Todo: add translation "Echelone du Type de Contrat"
+    count_type = fields.Integer(
```

Comment:
I'd use the more explicit `contract_type_count`.

Reply:
Good point!